core/src/main/java/io/neow3j/protocol/rx/JsonRpc2_0Rx.java

```diff
@@ -31,32 +28,50 @@ public JsonRpc2_0Rx(Neow3j neow3j, ScheduledExecutorService scheduledExecutorSer
-        });
+    /**
+     * Creates an observable that emits new block index as they are produced by the Neo blockchain. The observable
+     * polls the Neo node in the given {@code pollingInterval} to check for the latest block index and emits all
+     * indexes since the last time it polled.
+     *
+     * @param pollingInterval The polling interval in milliseconds.
+     * @return the block index observable.
+     */
+    public Observable<BigInteger> blockIndexObservable(long pollingInterval) {
+        return Observable.create(subscriber ->
+                new BlockIndexPolling().run(neow3j, subscriber, scheduledExecutorService, pollingInterval)
+        );
     }
 
-    public Observable<NeoGetBlock> replayBlocksObservable(BigInteger startBlock, BigInteger endBlock,
-            boolean fullTransactionObjects) {
-        return replayBlocksObservable(startBlock, endBlock, fullTransactionObjects, true);
+    /**
+     * Creates an observable that emits new blocks as they are produced by the Neo blockchain. The observable
+     * polls the Neo node in the given {@code pollingInterval} to check for the latest block and emits all
+     * blocks since the last time it polled.
+     *
+     * @param pollingInterval The polling interval in milliseconds.
+     * @return the block index observable.
+     */
+    public Observable<NeoGetBlock> blockObservable(boolean fullTransactionObjects, long pollingInterval) {
+        return blockIndexObservable(pollingInterval)
+                .flatMap(blockIndex -> neow3j.getBlock(blockIndex, fullTransactionObjects).observable());
     }
 
+    /**
+     * Creates an observable that emits blocks starting at {@code startBlockNumber} up to {@code endBlock} and then
+     * stops.
+     *
+     * @param startBlock             The block index at which to start.
+     * @param endBlock               The block index at which to stop.
+     * @param fullTransactionObjects If the full transactions objects should be included in the blocks.
+     * @param ascending              If the blocks should be emitted in ascending or descending order.
+     * @return the block index observable.
+     */
     public Observable<NeoGetBlock> replayBlocksObservable(BigInteger startBlock, BigInteger endBlock,
             boolean fullTransactionObjects, boolean ascending) {
-        // We use a scheduler to ensure this Observable runs asynchronously for users to be consistent with the other
-        // Observables.
         return replayBlocksObservableSync(startBlock, endBlock, fullTransactionObjects, ascending)
+                // We use a scheduler to run this Observable asynchronously
                 .subscribeOn(scheduler);
     }
 
-    private Observable<NeoGetBlock> replayBlocksObservableSync(BigInteger startBlock, BigInteger endBlock,
-            boolean fullTransactionObjects) {
-        return replayBlocksObservableSync(startBlock, endBlock, fullTransactionObjects, true);
-    }
```

Comment:
I removed this without side effects just because I wanted to strip down this class to the bear minimum for better maintainability. Whatever neow3j code used this method now simply uses the corresponding overload with an extra boolean argument at the end. Let me know if you don't agree at all with such cleanups.